src/BlTools.MssqlFluentSqlWrapper/FluentSqlCommand.cs

```diff
@@ -0,0 +1,520 @@
+        {
+            var res = new List<T>();
+            void ActionWrap(SqlDataReader reader, int resultNumber) => res.Add(itemBuilder(reader));
+
+            _dataReaderAction = ActionWrap;
+            Exec(ExecType.Reader);
+            return res;
+        }
+
+        public async Task<List<T>> ExecReadItemListAsync<T>(Func<SqlDataReader, T> itemBuilder)
+        {
+            var res = new List<T>();
+            void ActionWrap(SqlDataReader reader, int resultNumber) => res.Add(itemBuilder(reader));
+            _dataReaderAction = ActionWrap;
+            await ExecAsync(ExecType.Reader);
+            return res;
+        }
+
+        public T ExecReadItem<T>(Func<SqlDataReader, T> itemBuilder)
+        {
+            var result = default(T);
+            void ActionWrap(SqlDataReader reader, int resultNumber) => result = itemBuilder(reader);
+            _dataReaderAction = ActionWrap;
+            Exec(ExecType.Reader);
+            return result;
+        }
+
+        public async Task<T> ExecReadItemAsync<T>(Func<SqlDataReader, T> itemBuilder)
+        {
+            var result = default(T);
+            void ActionWrap(SqlDataReader reader, int resultNumber) => result = itemBuilder(reader);
+            _dataReaderAction = ActionWrap;
+            await ExecAsync(ExecType.Reader);
+            return result;
+        }
+
+        public string ExecReadXml()
+        {
+            var res = Exec(ExecType.XmlReader);
+            return res.XmlResult;
+        }
+
+        public async Task<string> ExecReadXmlAsync()
+        {
+            var res = await ExecAsync(ExecType.XmlReader);
+            return res.XmlResult;
+        }
+
+        public DataSet ExecFillDataSet(DataSet dataSet)
+        {
+            var result = Exec(ExecType.DataSet, new ExecResult { FillDataSetResult = dataSet });
+            return result.FillDataSetResult;
+        }
+
+        public async Task<DataSet> ExecFillDataSetAsync(DataSet dataSet)
+        {
+            var result = await ExecAsync(ExecType.DataSet, new ExecResult { FillDataSetResult = dataSet });
+            return result.FillDataSetResult;
+        }
+
+        public DataSet ExecFillDataSet()
+        {
+            return ExecFillDataSet(new DataSet());
+        }
+
+        public async Task<DataSet> ExecFillDataSetAsync()
+        {
+            return await ExecFillDataSetAsync(new DataSet());
+        }
+
+        #endregion
+
+        #region [ Command Post Execution Actions ]
+
+        public ExecStatus Status { get; private set; }
+
+        public Exception Error { get; private set; }
+
+        public T GetParamValue<T>(string name)
+        {
+            if (!_parameters.Contains(name))
+            {
+                throw new InvalidOperationException("No parameter with name [" + name + "] found");
+            }
+            var result = _parameters[name].Value;
+            return (T)(result == DBNull.Value ? default : result);
```

Comment:
`GetParamValue/Result` methods, `return (T)(result == DBNull.Value ? default : result);` vs `return (T)(result == DBNull.Value ? null : result);`
I'm not sure which way is better. Since we have generic, on the one hand `null` value can be not applicable (`default` is better), on the other hand cast of `null `to `T` will produce exception and `default `will be silent.
What do you think is right way ?

Reply:
I think `default` is correct. Casting null to ValueType would cause `NullReferenceException` which is not informative at all. And I don't think this is error situation. If we disallow `null` here - need to handle it explicitly, but I don't think we disallow...

Will change to `default` in Result<T>().

Reply:
Will also rename to GetResult<T>() since it's method, not property.